packages/components/src/interfaces/complex/Hero.ts

```diff
@@ -1,114 +1,309 @@
+        description: "The size of the hero banner",
+      }),
+    ),
+    dropdown: Type.Optional(HeroDropdownSchema),
+  }),
+])
+
+const HeroSideSchema = Type.Composite(
+  [
+    HeroInfoboxSchema,
+    HeroBackgroundImageSchema,
+    HeroKeyHighlightSchema,
+    Type.Object({
+      variant: Type.Literal("side"),
+    }),
+  ],
+  {
+    title: "Hero side variant",
+    groups: [
+      {
+        label: "Primary call-to-action",
+        fields: ["buttonLabel", "buttonUrl"],
+      },
+      {
+        label: "Secondary call-to-action",
+        fields: ["secondaryButtonLabel", "secondaryButtonUrl"],
+      },
+    ],
+  },
+)
+
+const HeroImageSchema = Type.Composite(
+  [
+    HeroBackgroundImageSchema,
+    HeroKeyHighlightSchema,
+    Type.Object({
+      variant: Type.Literal("image"),
+      dropdown: Type.Optional(HeroDropdownSchema),
+    }),
+  ],
+  {
+    title: "Hero image variant",
+  },
 )
+
+const HeroFloatingSchema = Type.Composite(
+  [
+    HeroInfoboxSchema,
+    HeroBackgroundImageSchema,
+    HeroKeyHighlightSchema,
+    Type.Object({
+      variant: Type.Literal("floating"),
+    }),
+  ],
+  {
+    title: "Hero floating variant",
+  },
+)
+
+const HeroCenterSchema = Type.Composite(
+  [
+    HeroHeadingSchema,
+    HeroButtonsSchema,
+    HeroBackgroundImageSchema,
+    HeroKeyHighlightSchema,
+    Type.Object({
+      variant: Type.Literal("center"),
+      dropdown: Type.Optional(HeroDropdownSchema),
+    }),
+  ],
+  {
+    title: "Hero centre variant",
+  },
+)
+
+const HeroGradientSchema = Type.Composite(
+  [
+    HeroHeadingSchema,
+    HeroButtonsSchema,
+    HeroBackgroundImageSchema,
+    HeroContentAlignmentSchema,
+    Type.Object({
+      variant: Type.Literal("gradient"),
+    }),
+  ],
+  {
+    title: "Hero gradient variant",
+  },
+)
+
+const HeroSplitSchema = Type.Composite(
+  [
+    HeroHeadingSchema,
+    HeroButtonsSchema,
+    HeroBackgroundImageSchema,
+    HeroContentAlignmentSchema,
+    Type.Object({
+      variant: Type.Literal("split"),
+      backgroundColor: Type.Optional(
+        Type.Union([Type.Literal("black"), Type.Literal("white")], {
+          title: "Hero background color",
+          description: "The background color of the hero banner",
+        }),
+      ),
+    }),
+  ],
+  {
+    title: "Hero split variant",
+  },
+)
+
+const HeroCopyLedSchema = Type.Composite(
+  [
+    HeroHeadingSchema,
+    HeroButtonsSchema,
+    Type.Partial(HeroBackgroundImageSchema),
+    HeroKeyHighlightSchema,
+    Type.Object({
+      variant: Type.Literal("copyled"),
+    }),
+  ],
+  {
+    title: "Hero copy-led variant",
+  },
+)
+
+const HeroFloatingImageSchema = Type.Composite(
+  [
+    HeroHeadingSchema,
+    HeroButtonsSchema,
+    HeroBackgroundImageSchema,
+    HeroKeyHighlightSchema,
+    Type.Object({
+      variant: Type.Literal("floatingimage"),
+    }),
+  ],
+  {
+    title: "Hero floating image variant",
+  },
+)
+
+export const HeroSchema = Type.Intersect(
```

Comment:
why do we need to export the schemas? just need to export the types right?

Reply:
Hmm the schemas are being referenced by IsomerComplexComponents, and we expose as part of a Map. That way, Studio can work directly with the component-level schemas instead of having to perform the extra step of extracting out from the general schema.